lint/src/main/java/com/firebaseui/lint/NonGlobalIdDetector.kt

```diff
@@ -0,0 +1,52 @@
+package com.firebaseui.lint
+
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.LayoutDetector
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.XmlContext
+import org.w3c.dom.Element
+
+/**
+ * Lint detector to find layout attributes that reference "@id" where they should reference
+ * "@+id" instead.
+ */
+class NonGlobalIdDetector : LayoutDetector() {
+    override fun getApplicableElements() = Detector.XmlScanner.ALL
```

Comment:
Let's specify the type here, `List<String>?` (I know it hurts 😄), since Kotlin is confused by opposing nullability annotations.

Reply:
I don't feel like it's necessary to add here.

Reply:
I hope I'm not being too much of a stickler, but here's my reasoning:
When it's a relatively small file like this, having a few platform dependent type warnings is fine, but should the file grow it just adds a whole bunch of unnecessary noise when you're trying to look at the scrollbar or `F2` through the file. Does that make sense?

PS: here's the error in case you turned it off:
![untitled](https://user-images.githubusercontent.com/9490724/34311638-8b2663f2-e713-11e7-901a-ccb6cca9ca4a.png)

Reply:
Yeah I'm aware of that. I'll change it.